test/camelCaseSyntax.spec.js

```diff
@@ -0,0 +1,226 @@
+      );
+    });
+  });
+
+  describe('in a clone', () => {
+    it('should succeed', () => {
+      expect.clone()(456).toBeGreaterThan(123);
+    });
+
+    it('should fail', () => {
+      expect(() => expect.clone()(123).toBeGreaterThan(456)).toThrow(
+        'expected 123 to be greater than 456'
+      );
+    });
+
+    it('should not support an assertion subsequently added to the original expect', () => {
+      const originalExpect = expect.clone();
+      const clonedExpect = originalExpect.clone();
+      originalExpect.addAssertion('<string> to foo', (expect, subject) =>
+        expect(subject).toEqual('foo')
+      );
+      expect(clonedExpect('foo').toFoo).notToBeAFunction();
+    });
+  });
+
+  describe('in a child', () => {
+    it('should succeed', () => {
+      expect.child()(456).toBeGreaterThan(123);
+    });
+
+    it('should fail', () => {
+      expect(() => expect.child()(123).toBeGreaterThan(456)).toThrow(
+        'expected 123 to be greater than 456'
+      );
+    });
+
+    it('should support an assertion subsequently added to the parent', () => {
+      const parentExpect = expect.clone();
+      const childExpect = parentExpect.child();
+      parentExpect.addAssertion('<string> to foo', (expect, subject) =>
+        expect(subject).toEqual('foo')
+      );
+      childExpect('foo').toFoo();
+    });
+  });
+
+  describe('with an assertion that is defined for the base of the given subject type', () => {
+    it('should succeed', () => {
+      function foo() {}
+      foo.bar = 123;
+      expect(foo).toHaveProperty('bar', 123);
+    });
+
+    it('should fail', () => {
+      function foo() {}
+      expect(() => expect(foo).toHaveProperty('bar', 123)).toThrow(
+        "expected function foo() {} to have property 'bar', 123"
+      );
+    });
+  });
+
+  describe('with an assertion that is not defined for the given subject type', () => {
+    it('should not expose a function', () => {
+      expect(expect(123).toContain).toBeUndefined();
+    });
+  });
+
+  describe('with a middle rocket assertion', () => {
+    it('should succeed', () => {
+      expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(3);
+    });
+
+    it('should fail', () => {
+      expect(() =>
+        expect([1, 2, 3]).whenPassedAsParametersTo(Math.max).toEqual(2)
+      ).toThrow('expected 3 to equal 2');
+    });
+  });
+
+  describe('with the expect.it equivalent', () => {
+    it('should succeed', () => {
+      expect({ foo: 123 }).toSatisfy({ foo: expect.toBeANumber() });
+    });
+
+    describe('and extra chaining with .and...', () => {
+      it('should succeed', () => {
+        expect({ foo: 123 }).toSatisfy({
+          foo: expect.toBeANumber().andToBeGreaterThan(100),
+        });
+      });
+
+      it('should fail', () => {
+        expect(() =>
+          expect({ foo: 123 }).toSatisfy({
+            foo: expect.toBeANumber().andToBeGreaterThan(200),
```

Comment:
I wonder if we could still attach these to a .it object and simply hand one out that has had the camelCaser run over it when we're in that mode?

Reason this occurs to me is that I think the .it. still serves a purpose syntactically in saying "and use the subject from the situation in which you are called.

Reply:
Yeah, I was a bit back and forth on that, decided that it looked fairly clean without any `it`. Could you spell out with some examples exactly which alternative(s) you're thinking about? `expect.it.toBeANumber()`? To me that reminded me too much of chai's weirdness where the dot notation and camel case are mixed -- `expect(foo).to.shallowDeepEqual(bar)` 🙀

Reply:
Hmm I see what you're getting at. Let me sit with this some more...

More thinking aloud, for the purposes of discussion, what do we do with the value returning form of expect.it? Perhaps the normal assertions should be callable directly, but we provide a .it() that is specifically meant to be used only as `.it(value => { ... })` ?

Reply:
Hmm, yeah, the `expect.it(fn)` case is interesting to consider. It shouldn't really be a problem to keep supporting it as-is.

You could argue that if we're going to migrate it over to something that's not backwards compatible (or if we want to add another variant of it that's easier on the eyes in a camel case context), we might as well switch to something completely different that isn't necessary an overloading of the `expect.it` replacement.